src/mlpack/methods/ann/activation_functions/elish_function.hpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file elish_function.hpp
+ * @author Bisakh Mondal
+ *
+ * Definition and implementation of the ELiSH function as described by
+ * Mina Basirat and Peter M. Roth
+ *
+ * For more information see the following paper
+ * 
+ * @code
+ * @misc{ELiSHActivation,
+ *    title={The Quest for the Golden Activation Function},
+ *    author={Mina Basirat and Peter M. Roth},
+ *    year={2018},
+ *    url={https://arxiv.org/pdf/1808.00783.pdf},
+ *    eprint={1808.00783},
+ *    archivePrefix={arXiv},
+ *    primaryClass={cs.NE} }
+ * @endcode
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ELISH_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ELISH_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The ELiSH function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     \frac{x}{1+e^{-x}} & : x \ge 0 \\
+ *     \frac{e^x-1}{1+e^{-x}} & : x < 0
+ *   \end{array}
+ * \right \\
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     \frac{1}{1+e^{-x}} + 
+ *              \frac{xe^{-x}}{ \left(1+e^{-x}}\right)^2} & : x \ge 0 \\
+ *     e^x-\frac{2}{1+e^x}+\frac{2}{\left(1+e^x}\right)^2} & : x < 0
+ *   \end{array}
+ * \right.
+ * @f}
+ */
+class ElishFunction
+{
+ public:
+ /**
+   * Computes the ELiSH function.
+   *
+   * @param x Input data.
+   * @return f(x).
+   */
+  static double Fn(const double x)
+  {
+    if (x < 0.0) {
+        return (std::exp(x) - 1) / (1 + std::exp(-x));
+    }
+    return x / (1 + std::exp(-x));
+  }
+
+  /**
+   * Computes the ELiSH function.
+   *
+   * @param x Input data.
+   * @param y The resulting output activations.
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Fn(const InputVecType& x, OutputVecType& y)
+  {
+    y.set_size(size(x));
+
+    for (size_t i = 0; i < x.n_elem; i++)
```

Comment:
Hey @bisakhmondal I think we should use 
```
    if (x < 0.0) {
        y = (arma::exp(x) - 1) / (1 + arma::exp(-x));
    }
    y = x / (1 + arma::exp(-x));
  }
``` 
Instead of using for loop. @zoq I am little bit confused about this can you tell which one will be a better approach.

Reply:
Agreed, would be nice to use `arma` here, in some cases it's faster.

Reply:
Done. Changed to `arma`.

Reply:
I am not sure I am seeing the change. I think what @himanshupathak21061998 meant was rather than using `for` loop we can write this as `y =  x / (1 + arma::exp(-x));`. Let me know what you think. Thanks.